marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/managers/EntitySearchManager.java

```diff
@@ -100,7 +112,7 @@ public StringHandle search(SearchQuery searchQuery) {
         }
         catch (MarkLogicServerException e) {
             // If there are no entityModels to search, then we expect an error because no search options will exist
-            if(searchQuery.getQuery().getEntityTypeIds().isEmpty() || modelManager.getModels().size() == 0) {
+            if (searchQuery.getQuery().getEntityTypeIds().isEmpty() || modelManager.getModels().size() == 0) {
```

Comment:
Looking at this, it does seem like there's a chance for a null pointer. How about "searchQuery.hasEntityTypeIds()"? That results in less work for the controller, and that method can be coded to ensure an NPE can't occur.

Reply:
Since the models have been designed to initialize these objects in the constructor it won't throw an NPE.